sources/packages/backend/apps/api/src/services/institution/institution.service.ts

```diff
@@ -960,20 +967,34 @@ export class InstitutionService extends RecordDataModelService<Institution> {
    * has access to student data of given student.
    * @param institutionId institution.
    * @param studentId student.
+   * @param options options for the query:
+   * - `applicationId` application id.
    * @returns value which specifies if the institution
    * has access to student data of given student.
    */
   async hasStudentDataAccess(
     institutionId: number,
     studentId: number,
+    options?: {
+      applicationId?: number;
+    },
   ): Promise<boolean> {
-    const institutionStudentDataAccess = await this.applicationRepo.findOne({
-      select: { id: true },
-      where: {
+    let findOptions: FindOptionsWhere<Application> = {
+      student: { id: studentId },
+      location: { institution: { id: institutionId } },
+      applicationStatus: Not(ApplicationStatus.Overwritten),
+    };
+    if (options?.applicationId) {
```

Comment:
I believe that the below code would work in the same way. Ca you please test it?
```ts
const institutionStudentDataAccess = await this.applicationRepo.findOne({
  select: { id: true },
  where: {
    student: { id: studentId },
    location: { institution: { id: institutionId } },
    applicationStatus: Not(ApplicationStatus.Overwritten),
    id: options?.applicationId,
  },
  cache: true,
  });
return !!institutionStudentDataAccess;
```

Reply:
actually, I tried this first, but it was throwing me below error, when `application.id` was undefined,
![image](https://github.com/bcgov/SIMS/assets/77353155/0cd18f1c-f57e-4a09-b1d1-46a2d4b5c3f6)

Reply:
as discussed updated